--- origsrc/rcairo-1.10.0/ext/cairo/extconf.rb	2010-09-12 02:21:12.000000000 -0500
+++ src/rcairo-1.10.0/ext/cairo/extconf.rb	2010-11-01 02:40:27.234550200 -0500
@@ -40,6 +40,11 @@ checking_for(checking_message("Win32 OS"
     PKGConfig.add_path(pkg_config_dir.to_s)
     PKGConfig.set_override_variable("prefix", binary_base_dir.to_s)
     true
+  when /cygwin/
+    import_library_name = "libruby-#{module_name}.a"
+    $DLDFLAGS << " -Wl,--out-implib=#{import_library_name}"
+    $cleanfiles << import_library_name
+    false
   else
     false
   end
